app/src/main/java/com/example/practice/MainActivityViewModel.kt

```diff
@@ -35,7 +35,7 @@ class MainActivityViewModel : ViewModel() {
     fun fetchUserList(excludingUserId: String = ""){
 
         // Call Network Api to get user data
-        RetrofitAPI().fetchUsersList(excludingUserId, {
+        API.create().fetchUsersList(excludingUserId, {
 
             // Perform task in coroutine as operation it time consuming
             viewModelScope.launch(Dispatchers.IO) {
```

Comment:
Line 78 contains the following statement: `it?.id?.equals(excludingUserId)!!`
What scenario could cause application crash at this line? How would you avoid it?

Reply:
@nets-dev code will be expected to crash when 'id' will set to 'null'.
Fix : 
Check whether 'id' is null and if 'id' is null return 'false' or return 'id.equals(excludingUserId)'